core/server/master/src/main/java/alluxio/master/file/meta/TtlBucketList.java

```diff
@@ -89,8 +89,19 @@ public void insert(Inode<?> inode) {
       // start time of this interval should be (ttlEndTimeMs / interval) * interval.
       long interval = TtlBucket.getTtlIntervalMs();
       bucket = new TtlBucket(interval == 0 ? ttlEndTimeMs : ttlEndTimeMs / interval * interval);
-      mBucketList.add(bucket);
+      synchronized (mBucketList) {
```

Comment:
Consider this scenario: two file with almost same creation time and TTL are now being inserted into the bucket in two separate threads. Suppose currently both threads get a null after calling `TtlBucketList#getBucketContaining`, then they will both initialize a new `TtlBucket`, and both new `TtlBucket`s share the same TtlIntervalStartTimeMs. However, only one of them will be added into `mBucketList` for the reason that the two buckets are equal to each other, and the `ConcurrentSkipListSet` will discard the second bucket when trying to add it.